lib/src/rules/avoid_returning_this.dart

```diff
@@ -99,9 +96,9 @@ class _Visitor extends SimpleAstVisitor<void> {
     if (body is BlockFunctionBody) {
       var returnStatements = body.block
           .traverseNodesInDFS(excludeCriteria: _isFunctionExpression)
-          .where(_isReturnStatement);
+          .whereType<ReturnStatement>();
       if (returnStatements.isNotEmpty && returnStatements.every(_returnsThis)) {
-        rule.reportLintForToken(node.name2);
+        rule.reportLint(returnStatements.first);
```

Comment:
I'd prefer narrowing it further to reporting the diagnostic on the `this`.

Reply:
Done!